### PR TITLE
Download daml-finance dependencies automatically

### PR DESCRIPTION
**Pull request by bartfaitamas**

### CODE REVIEW

File: src/main/java/com/daml/extensions/damlmavenplugin/DependencyUtils.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+public abstract class DependencyUtils {
+
+    static Pattern dependencyPattern = Pattern.compile("^((contingent-claims|daml-finance)-[-\\w]+)-([.\\d]+).dar$");
+
+    public static DependencyUtils defaultInstance(Log log) {
+        return new DependencyUtils() {
+            @Override
+            protected void downloadDependency(URL url, Path target) throws IOException {
+                log.info(String.format("Download daml dependency %s -> %s", url, target));
```
Comment (gaborh-da) [inline src/main/java/com/daml/extensions/damlmavenplugin/DependencyUtils.java]: Downloading

File: src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java [hunk 2]
```diff
@@ -4,18 +4,35 @@
...
-            getLog().info(scanner.nextLine());
+        try (Scanner scanner = new Scanner(in)) {
+            while (scanner.hasNextLine()) {
+                getLog().info(scanner.nextLine());
+            }
+        }
+    }
+
+    protected Path getDamlYamlFile() throws MojoFailureException {
+        Path cwd = Paths.get("").toAbsolutePath();
```
Comment (gaborh-da) [inline src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java]: get(".") I think is the same and would be a little be easier to read/understand

File: src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java [hunk 3]
```diff
@@ -4,18 +4,35 @@
...
+        while (cwd != null) {
+            Path damlYaml = cwd.resolve("daml.yaml");
```
Comment (gaborh-da) [inline src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java]: it does not make much diff but I like to have private static-s for such "magic strings" (if we can say it is one)

Comment (gaborh-da) [reply src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java]: (this stands for the delimiters and all the string literals)

**Review by gaborh-da — APPROVED** [review]
approved, added a couple of nits

File: src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java (hunk 2 above)
Comment (bartfaitamas) [reply src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java]: It produces a path "/foo/./bar", which is confusing for the user

File: src/main/java/com/daml/extensions/damlmavenplugin/DependencyUtils.java (hunk 1 above)
Comment (bartfaitamas) [reply src/main/java/com/daml/extensions/damlmavenplugin/DependencyUtils.java:28]:
```suggestion
                log.info(String.format("Downloading daml dependency %s -> %s", url, target));
```

File: src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java (hunk 3 above)
Comment (bartfaitamas) [reply src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java]: Made some more improvements, but didn't forget to move the string literal to a constant

**Pull request merged**